internal/pkg/scaffold/crd.go

```diff
@@ -58,58 +63,51 @@ func (s *CRD) getFS() afero.Fs {
 
 func (s *CRD) GetInput() (input.Input, error) {
 	if s.Path == "" {
-		fileName := fmt.Sprintf("%s_%s_%s_crd.yaml",
-			s.Resource.GoImportGroup,
-			strings.ToLower(s.Resource.Version),
-			s.Resource.LowerKind)
-		s.Path = filepath.Join(CRDsDir, fileName)
+		s.Path = crdPathForResource(CRDsDir, s.Resource)
 	}
 	return s.Input, nil
 }
 
+func crdPathForResource(dir string, r *Resource) string {
+	lv := strings.ToLower(r.Version)
+	file := fmt.Sprintf("%s_%s_%s_crd.yaml", r.GoImportGroup, lv, r.LowerKind)
+	return filepath.Join(dir, file)
+}
+
+type crdOutputRule struct {
+	fs afero.Fs
+}
+
+var _ genall.OutputRule = crdOutputRule{}
+
+// Open is meant to be used to generate a CRD manifest in memory at path.
+func (o crdOutputRule) Open(_ *loader.Package, path string) (io.WriteCloser, error) {
```

Comment:
Where is this used?

Reply:
The CRD generator will call `o.Open()` when writing the CRD manifest because it is an [`OutputRule`](https://github.com/operator-framework/operator-sdk/pull/1642/files#diff-88638833fc541d7d24b44b34835b0fb4R168).